cmake/modules/CheckLinkerDynamicRelocationFlags.cmake

```diff
@@ -0,0 +1,11 @@
+include(CheckCXXCompilerFlag)
+
+function(check_linker_supports_flag flag_to_test var_name)
+  set(CMAKE_REQUIRED_FLAGS "${flag_to_test}")
+  check_cxx_compiler_flag("" LINKER_SUPPORTS_NOEXTERN_PROTECTED_DATA)
+  if(LINKER_SUPPORTS_NOEXTERN_PROTECTED_DATA)
```

Comment:
I think I should clarify the problem. For example:
```cmake
cmake_minimum_required(VERSION 2.8.12)                                            
include(CheckCXXCompilerFlag)                                                     
                                                                               
check_cxx_compiler_flag("-static" RESULT)                                         
message("RESULT_STATIC: ${RESULT}")                                               
check_cxx_compiler_flag("-FOO-BAR" RESULT)                                        
message("RESULT_FOOBAR: ${RESULT}")                                               
```
This `CMakeLists.txt` outputs:
```
-- Performing Test RESULT
-- Performing Test RESULT - Success
RESULT_STATIC: 1
RESULT_FOOBAR: 1
```
The second `check_cxx_compiler_flag()` is no-op and just returns cached `RESULT` value.
Generated `CMakeCache.txt` contains
```
//Test RESULT                                                                     
RESULT:INTERNAL=1                                                                 
```